### PR TITLE
P/brentpicasso/brent faster trackmap update

### PR DESCRIPTION
**Pull request by brentpicasso**
continuing the PR. Added some fixes when updating tracks.

### CODE REVIEW

**Comment by brentpicasso** [comment]
The initial pause while downloading tracks needs a user message, otherwise looks pretty good.

File: autosportlabs/racecapture/tracks/trackmanager.py [hunk 1]
```diff
@@ -5,138 +5,127 @@
...
-                    sectorPoints.append(GeoPoint.fromPoint(point[0], point[1]))
-            self.sectorPoints = sectorPoints
-            if not self.shortId > 0:
-                raise Warning("Could not parse trackMap: shortId is invalid") 
+
+    def from_dict(self, track_dict):
+        """Populate this object's values with values from a dict, either from RCL's API or a file
+        """
+
+        self.start_finish_point = GeoPoint.fromPointJson(track_dict.get('start_finish'))
```
Comment (stieg) [inline autosportlabs/racecapture/tracks/trackmanager.py]: Any need to sanitize these values or check that they are valid/within scope?

File: autosportlabs/racecapture/tracks/trackmanager.py [hunk 2]
```diff
@@ -5,138 +5,127 @@
...
+        self.name = track_dict.get('name', self.name)
+        self.configuration = track_dict.get('configuration', self.configuration)
+        self.length = track_dict.get('length', self.length)
+        self.track_id = track_dict.get('id')
+
+        map_points_array = track_dict.get('track_map_array')
+
+        if map_points_array:
+            for point in map_points_array:
+                self.map_points.append(GeoPoint.fromPoint(point[0], point[1]))
```
Comment (stieg) [inline autosportlabs/racecapture/tracks/trackmanager.py]: Is this sane with the last point on the list?

File: autosportlabs/racecapture/tracks/trackmanager.py [hunk 3]
```diff
@@ -5,138 +5,127 @@
...
+
+        sector_array = track_dict.get('sector_points')
+
+        if sector_array:
```
Comment (stieg) [inline autosportlabs/racecapture/tracks/trackmanager.py]: Duplicated code here but it is small.  Consider cleaning it up

File: autosportlabs/racecapture/tracks/trackmanager.py (hunk 2 above)
Comment (stieg) [inline autosportlabs/racecapture/tracks/trackmanager.py]: NVM.  Didn't realize this was lat/lon

File: autosportlabs/racecapture/tracks/trackmanager.py [hunk 4]
```diff
@@ -144,236 +133,292 @@ def setTracksUserDir(self, path):
...
+
+    def fetch_venue_list(self, full_response=False):
+        """Fetches all venues from RCL's API and returns them as an array of dicts. RCL's API normally returns minimal
+        object information when listing multiple objects. The 'full_response' arg tells this function to expand
+        all objects to contain all their data. This allows us to quickly get basic information about tracks or pull
+        down everything if we have no tracks locally.
+        """
+
+        total_venues = None
+        next_uri = self.RCP_VENUE_URL + "?start=0&per_page=100"
```
Comment (stieg) [inline autosportlabs/racecapture/tracks/trackmanager.py]: Might be wise to put this URI generator into its own function.  Good for testing and general sanity.

**Comment by stieg** [comment]
Reviewed.  Looks pretty good.  Also what testing do we currently have on this (if any)?

File: autosportlabs/racecapture/tracks/trackmanager.py (hunk 1 above)
Comment (brentpicasso) [inline autosportlabs/racecapture/tracks/trackmanager.py]: Maybe. but they come from a trusted source so the risk is fairly minimal.

**Pull request merged**